scipy/linalg/_matfuncs_expm.pyx.in

```diff
@@ -146,9 +146,12 @@ def pick_pade_structure(cnp.ndarray[ndim=3, dtype=numpy_lapack_t] Am):
     # m = 3
     # -------
     # 1-norm of A ** 7
-    for i in range(6):
-        np.matmul(absA, work_arr, out=work_arr)
-    temp = np.max(work_arr)
+    np.matmul(absA, work_arr, out=work_arr)
+    np.matmul(absA, work_arr, out=work_arr)
+    absA_pow4 = np.matmul(absA, work_arr)
+    del absA
```

Comment:
You don't need to create and delete arrays here. Just overwrite `absA` or use `Am[4, :, :]` for temporaries, deleting big arrays are quite expensive.